Lib/test/clinic.test.c

```diff
@@ -6436,4 +6438,193 @@ test_deprecate_positional_pos2_len3_with_kwdonly_impl(PyObject *module,
                                                       PyObject *c,
                                                       PyObject *d,
                                                       PyObject *e)
-/*[clinic end generated code: output=383d56b03f7c2dcb input=154fd450448d8935]*/
+/*[clinic end generated code: output=0c7e90700b1ab0dd input=154fd450448d8935]*/
+
+
+/*[clinic input]
+@classmethod
+Test.__new__
+    * [from 3.14]
+    a: object
+The deprecation message should use the class name instead of __new__.
+[clinic start generated code]*/
+
+PyDoc_STRVAR(Test__doc__,
+"Test(a)\n"
+"--\n"
+"\n"
+"The deprecation message should use the class name instead of __new__.\n"
+"\n"
+"Note: Passing positional arguments to Test() is deprecated. Parameter\n"
+"\'a\' will become a keyword-only parameter in Python 3.14.\n"
+"");
+
+static PyObject *
+Test_impl(PyTypeObject *type, PyObject *a);
+
+static PyObject *
+Test(PyTypeObject *type, PyObject *args, PyObject *kwargs)
+{
+    PyObject *return_value = NULL;
+    #if defined(Py_BUILD_CORE) && !defined(Py_BUILD_CORE_MODULE)
+
+    #define NUM_KEYWORDS 1
+    static struct {
+        PyGC_Head _this_is_not_used;
+        PyObject_VAR_HEAD
+        PyObject *ob_item[NUM_KEYWORDS];
+    } _kwtuple = {
+        .ob_base = PyVarObject_HEAD_INIT(&PyTuple_Type, NUM_KEYWORDS)
+        .ob_item = { &_Py_ID(a), },
+    };
+    #undef NUM_KEYWORDS
+    #define KWTUPLE (&_kwtuple.ob_base.ob_base)
+
+    #else  // !Py_BUILD_CORE
+    #  define KWTUPLE NULL
+    #endif  // !Py_BUILD_CORE
+
+    static const char * const _keywords[] = {"a", NULL};
+    static _PyArg_Parser _parser = {
+        .keywords = _keywords,
+        .fname = "Test",
+        .kwtuple = KWTUPLE,
+    };
+    #undef KWTUPLE
+    PyObject *argsbuf[1];
+    PyObject * const *fastargs;
+    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    PyObject *a;
+
+    // Emit compiler warnings when we get to Python 3.14.
+    #if PY_VERSION_HEX >= 0x030e00C0
+    #  error \
+            "In clinic.test.c, update parameter(s) 'a' in the clinic input of" \
+            " Test() to be keyword-only."
```

Comment:
I am not sure that is an improvement. `Test` is a class, not function, there is no clinic input for it (or at least not what have parameters). Even if you guess that it is a reference to a constructor, it has two constructors. I think that `Test.__new__` would be better.

Also, in clinic input to which the message refers it is named without parentheses.

Reply:
Hm, yes for the preprocessor message it is actually best to keep the original. I am not so sure about the deprecation message though. Will it not be better for users with this improvement?

Reply:
The user usually do not call `__init__` or `__new__` directly, so it is better to use the class name there. And parentheses are at the right place there. The runtime warning message is consistent with an error message raised when pass wrong keyword argument to a function/class.

Reply:
So I'll keep that change. Marking this as resolved.